File: dotnet/In3/Crypto/CurveType.cs
namespace In3.Crypto
{
    /// <summary>
    /// Enum to be used along with the methods of <see cref="Crypto.Signer" />.
    /// </summary>
    public enum CurveType : int
    {  
        /// <summary>Sign with ecdsa</summary>
        Ecdsa   = 1,
        /// <summary>Use ed25519 curve</summary>
        Ed25519 = 2,
    }
}

File: java/src/in3/utils/CurveType.java
package in3.utils;

/*
 * Constants for the type of Signature
 */
public enum CurveType {
  /**< sign with ecdsa */
  ecdsa(1),
  /**< use ed25519 curve */
  ed25519(2);

  public final int val;

  private CurveType(int val) {
    this.val = val;
  }

  private int getValue() {
    return this.val;
  }

  public static CurveType getEnum(int enumVal) {
    for (CurveType v : values())
      if (v.getValue() == enumVal) return v;
    return null;
  }
}
